### PR TITLE
[Compute] New command group az vm image terms

### PR DESCRIPTION
**Pull request by qwordy**
Issue #9597
Excited to announce 3 new commands!
az vm image terms accept
az vm image terms cancel
az vm image terms show

vm image accept-terms, 3234 requests in last 30 days.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/_help.py [hunk 1]
```diff
@@ -1405,6 +1405,35 @@
     crafted: true
 """
 
+helps['vm image terms'] = """
+type: group
+short-summary: Manage Azure Marketplace terms.
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py]: Marketplace Image terms

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -1441,6 +1441,74 @@ def accept_market_ordering_terms(cmd, urn=None, publisher=None, offer=None, plan
 # endregion
 
 
+def _terms_prepare(cmd, urn, publisher, offer, plan):
+    usage_err = 'usage error: --plan STRING --offer STRING --publish STRING |--urn STRING'
+    if urn:
+        if any([publisher, offer, plan]):
+            raise CLIError(usage_err)
+        publisher, offer, _, _ = urn.split(':')
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: add try/except for split(':')

**Comment by jiasli** [comment]
The code looks good. But I prefer to create a new command group called `marketplace` and put `terms` there.

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: We can also add a validator to do argument checks.

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 3]
```diff
@@ -1441,6 +1441,74 @@ def accept_market_ordering_terms(cmd, urn=None, publisher=None, offer=None, plan
...
+    :return:
+    """
+    return _accept_cancel_terms(cmd, urn, publisher, offer, plan, True)
+
+
+def cancel_terms(cmd, urn=None, publisher=None, offer=None, plan=None):
+    """
+    Cancel terms.
+    :param cmd:cmd
+    :param urn:URN, in format of 'publisher:offer:sku:version'. If specified, other argument values can be omitted
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: well. re-think about this. Same help information appear in three places.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: Yes. But it's not code. I can't use variables to concatenate doc.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: Updated

**Pull request merged**